### PR TITLE
Fix name clash of RNG with shared memory

### PR DESCRIPTION
**Pull request by zasdfgbnm**
RNG uses
```C++
auto offset = philox_args.captured_ ? ....
```
Shared memory uses
```C++
unsigned offset = 0;
```
So when a scheduler tries to schedule RNG with shared memory, compilation will fail
```C++
CUDA NVRTC compile error: __tmp_kernel1.cu(6686): error: "offset" has already been declared in the current scope

__tmp_kernel1.cu(6679): warning #177-D: variable "offset" was declared but never referenced

1 error detected in the compilation of "__tmp_kernel1.cu".
```

This is NOT a new bug introduced in https://github.com/csarofeen/pytorch/pull/1888, but instead, both names have been there for a long time. It is just I recently added both a test of RNG with broadcasting and a transpose scheduler.

In general, we should avoid common names like `offset`, we should use `how_you_will_use_it_offset` instead.

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
Still running tests.

File: torch/csrc/jit/codegen/cuda/codegen.cpp [hunk 1]
```diff
@@ -719,7 +719,7 @@ class CudaKernelGenerator : private OptOutConstDispatch {
         indent() << "if (rng_subseq != subseq" << uop->name()
                  << " || rng_offset != offset" << uop->name() << ") {\n";
         indent() << "  rng_result = philox(philox_args.seed_, subseq"
-                 << uop->name() << ", offset / 4 + offset" << uop->name()
+                 << uop->name() << ", philox_offset / 4 + offset" << uop->name()
```
Comment (shmsong) [inline torch/csrc/jit/codegen/cuda/codegen.cpp]: Do we need to rename the `offset` here too?

Comment (shmsong) [reply torch/csrc/jit/codegen/cuda/codegen.cpp]: I see. This one is suffixed by the uop name so would be ok.

**Review by shmsong — APPROVED** [review]
LGTM. Thanks so much for fixing this!

File: torch/csrc/jit/codegen/cuda/codegen.cpp (hunk 1 above)
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/codegen.cpp]: The second offset is `offset123`, so not necessary. But it sounds like a good idea to rename it as well. Will push an update.

**Pull request merged**